node/tools/src/main.rs

```diff
@@ -2,14 +2,41 @@
 //! manages communication between the actors. It is the main executable in this workspace.
 use anyhow::Context as _;
 use clap::Parser;
-use std::{fs, io::IsTerminal as _, path::PathBuf};
+use std::{fs, io::IsTerminal as _, path::PathBuf, str::FromStr};
 use tracing::metadata::LevelFilter;
 use tracing_subscriber::{prelude::*, Registry};
 use vise_exporter::MetricsExporter;
 use zksync_concurrency::{ctx, scope};
+use zksync_consensus_crypto::read_required_text;
+use zksync_consensus_roles::node;
 use zksync_consensus_tools::ConfigPaths;
 use zksync_consensus_utils::no_copy::NoCopy;
 
+/// Utility struct to parse json value from cli arg
+#[derive(Debug, Clone)]
+struct NodeAddresses(Vec<(node::PublicKey, std::net::SocketAddr)>);
```

Comment:
please move to config crate and implement `ProtoFmt` for it instead. Deduplicate the code with `ProtoFmt` implementation for `AppConfig`.